cypress/support/cypress-commands/lifecycle.js

```diff
@@ -198,13 +198,18 @@ Cypress.Commands.add('validateLifecycleHistoryAndEvents', (state, appId) => {
       const lifecycleEventRequirementId = scenarioRequirement.find((req) =>
         req.hasOwnProperty('event')
       );
+      const appHistoryPrevious = UTILS.getEnvVariable(CONSTANTS.APP_LIFECYCLE_HISTORY);
+      const appHistoryCount = appHistory.length - appHistoryPrevious.length;
 
       // Lifecycle event validation
-      if (lifecycleEventRequirementId && lifecycleEventRequirementId.event) {
-        const appHistoryPrevious = UTILS.getEnvVariable(CONSTANTS.APP_LIFECYCLE_HISTORY);
-        const appHistoryCount = appHistory.length - appHistoryPrevious.length;
+      if (condition == CONSTANTS.STAY) {
```

Comment:
Why use `constants.stay`?

Why not, in the glue use `(be|stay)` to set a boolean flag called `isEventsExpected`
Then pass that into `validateLifecycleHistoryAndEvents`
Much cleaner and improved readability

Reply:
added a new Boolean variable as mentioned